server/container/guice/guice-common/src/test/java/org/apache/james/AggregateJunitExtension.java

```diff
@@ -0,0 +1,67 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.apache.james.util.Runnables;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.Lists;
+
+public class AggregateJunitExtension implements RegistrableExtension {
+
+    private final List<? extends RegistrableExtension> registrableExtensions;
+
+    public AggregateJunitExtension(List<? extends RegistrableExtension> registrableExtensions) {
+        this.registrableExtensions = registrableExtensions;
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) {
+        Runnables.runParrallelStream(registrableExtensions
```

Comment:
As said during the daily, this parrallel init of extension seems dangerous because it doesn't respect dependency hierarchy

Reply:
My option: We can keep the parallel build. It saves us time(don't know how much time). Anyway basically,`registrableExtensions` should be independent. If there is an`ABCRegistrableExtension` depends on `XYZRegistrableExtension` at `beforeEach` step. I would write a new `AggregateRegistrableExtension` and compose two above extensions into one aggregate extension to make it run `beforeEach` in order.